src/EditorFeatures/Core/Implementation/InlineRename/InlineRenameSession.cs

```diff
@@ -260,11 +260,21 @@ private void InitializeOpenBuffers(SnapshotSpan triggerSpan)
             RenameTrackingDismisser.DismissRenameTracking(_workspace, _workspace.GetOpenDocumentIds());
         }
 
+        private const string ContainedLanguageMarker = nameof(ContainedLanguageMarker);
+
         private bool TryPopulateOpenTextBufferManagerForBuffer(ITextBuffer buffer)
         {
             AssertIsForeground();
             VerifyNotDismissed();
 
+            // TS creates generated documents to back script blocks in razor generated files.
+            // These files are opened in the roslyn workspace but are not valid to rename
+            // as they are not proper buffers.  So we exclude any buffer that is marked as a contained language.
+            if (buffer.Properties.TryGetProperty<bool>(ContainedLanguageMarker, out var markerValue) && markerValue)
```

Comment:
Is there no way that this logic can go in the `SupportsRename` system somewhere? It looks like there is already some logic for [Razor](http://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/Venus/ContainedDocument.cs,89) and [contained documents](http://sourceroslyn.io/#Microsoft.VisualStudio.LanguageServices/Implementation/VisualStudioSupportsFeatureService.cs,103) but I have no idea if those are relevant at all.

Reply:
good call, let me see if I can do something like that.

Reply:
Yes, these virtual TS files never use our rename service, so it's fine to have them not support rename.